src/t_stream.c

```diff
@@ -954,9 +976,23 @@ static int streamParseAddOrTrimArgsOrReply(client *c, streamAddTrimArgs *args, i
         } else if (xadd && !strcasecmp(opt,"nomkstream")) {
             args->no_mkstream = 1;
         } else if (xadd) {
+            /* At this point, we check if the ID is in the form of "ms-*". */
+            sds arg = c->argv[i]->ptr;
+            args->seq_given = (arg[sdslen(arg)-1] != '*');
+            if (!args->seq_given) {
+                /* A dirty hack so we can still parse the ID. We'll revert the
+                 * change immediately. */
+                arg[sdslen(arg)-1] = '0';
+            }
+            
             /* If we are here is a syntax error or a valid ID. */
-            if (streamParseStrictIDOrReply(c,c->argv[i],&args->id,0) != C_OK)
+            int res = streamParseStrictIDOrReply(c,c->argv[i],&args->id,0);
+            if (!args->seq_given) {
+                arg[sdslen(arg)-1] = '*';
+            }
+            if (res != C_OK) {
                 return -1;
+            }
             args->id_given = 1;
```

Comment:
you now have to fix the code added in #9599

Reply:
Resolved via 318c4868d